### PR TITLE
Document CNS Support

### PR DESCRIPTION
**Pull request by bmcelvee**
FYI @aheslin @tmorriso-rh

### CODE REVIEW

**Comment by tmorriso-rh** [comment]
Thanks, Brandi! 👍  Please let me know if you need any assistance with this PR.

**Comment by bmcelvee** [comment]
@tmorriso-rh Do you think this PR is still relevant with the updates that are in progress this week?

**Comment by tmorriso-rh** [comment]
@bmcelvee Hey Brandi, AFAIK, this PR is still relevant. The work happening in PR https://github.com/openshift/openshift-docs/pull/8149 does not include any of the files in PR 6655. I think @jarrpa needs to review PR 6655 to see if the information has changed. Thanks.

**Comment by bmcelvee** [comment]
Sounds good. Thanks, @tmorriso-rh! :cherry_blossom:

File: install_config/aggregate_logging.adoc [hunk 1]
```diff
@@ -655,8 +655,7 @@ specified, it is created with a request for that size.
 
 [WARNING]
 ====
-Using NFS storage as a volume or a persistent volume (or via NAS such as
-Gluster) is not supported for Elasticsearch storage, as Lucene relies on file
+Using NFS storage as a volume or a persistent volume is not supported for Elasticsearch storage, as Lucene relies on file
```
Comment (jarrpa) [inline install_config/aggregate_logging.adoc]: Is there a reason these two lines were merged? Seems inconsistent with the rest of the formatting.

File: install_config/aggregate_logging.adoc [hunk 2]
```diff
@@ -676,6 +675,11 @@ This loopback must be maintained manually outside of {product-title}, on the
 node. You must not maintain it from inside a container.
 ====
 
+You can use Container-Native Storage (CNS) 3.6 to enable Gluster Block Storage
```
Comment (jarrpa) [inline install_config/aggregate_logging.adoc]: A few things:
* There is no such thing as CNS in the upstream. If possible, please see if you can make use of the variables I'm introducing in #8149, namely `{gluster-native}`, as that will correctly present wither "Containerized GlusterFS" or "Container-Native Storage" depending on the doc site.
* In my mind there's no need to mention a specific version. If you must, I request that you instead say "3.6 and onward" and that you encapsulate it in an `ifdef::openshift-enterprise[]` block.
* I've been stylizing it as `gluster-block` in my documentation. I think that's correct? If there's any official stylization for it I'd be happy to know and correct my docs if needed.

File: install_config/aggregate_logging.adoc [hunk 3]
```diff
@@ -676,6 +675,11 @@ This loopback must be maintained manually outside of {product-title}, on the
...
+as a backend for logging with ElasticSearch. For more information, see
```
Comment (jarrpa) [inline install_config/aggregate_logging.adoc]: I believe it's supposed to be `Elasticsearch`, as you had it above.

File: install_config/aggregate_logging.adoc [hunk 4]
```diff
@@ -676,6 +675,11 @@ This loopback must be maintained manually outside of {product-title}, on the
...
+link:https://access.redhat.com/documentation/en-us/red_hat_gluster_storage/3.3/html/container-native_storage_for_openshift_container_platform/apes03[Enabling
```
Comment (jarrpa) [inline install_config/aggregate_logging.adoc]: This link should point to the example I'm writing on how to structure the Advanced Installer inventory file to enable this. Now sure how that should be handled, however.

File: install_config/cluster_metrics.adoc [hunk 5]
```diff
@@ -121,6 +121,10 @@ If you would like to use xref:../install_config/persistent_storage/dynamically_p
 xref:../install_config/cluster_metrics.adoc#metrics-ansible-variables[variable]
 to `dynamic` in the inventory file.
 
+You can set this variable with Container-Native Storage (CNS) 3.6 to enable Gluster
```
Comment (jarrpa) [inline install_config/cluster_metrics.adoc]: As before, no such thing as CNS in the upstream.

File: install_config/cluster_metrics.adoc [hunk 6]
```diff
@@ -121,6 +121,10 @@ If you would like to use xref:../install_config/persistent_storage/dynamically_p
...
+Block Storage as a backend for metrics with Cassandra. For more information, see
+link:https://access.redhat.com/documentation/en-us/red_hat_gluster_storage/3.3/html/container-native_storage_for_openshift_container_platform/apes03[Enabling Gluster Block Storage as Backend for Metrics].
```
Comment (jarrpa) [inline install_config/cluster_metrics.adoc]: Same thing for this link as with the Logging one.

File: install_config/install/advanced_install.adoc [hunk 7]
```diff
@@ -1565,6 +1565,10 @@ xref:../../install_config/persistent_storage/dynamically_provisioning_pvs.adoc#i
 openshift_metrics_cassandra_storage_type=dynamic
 ----
 
+You can set this variable with CNS 3.6 to enable Gluster
+Block Storage as a backend for logging with ElasticSearch. For more information,
+see link:https://access.redhat.com/documentation/en-us/red_hat_gluster_storage/3.3/html/container-native_storage_for_openshift_container_platform/apes02[Enabling Gluster Block Storage as Backend for Logging].
+
```
Comment (jarrpa) [inline install_config/install/advanced_install.adoc]: This is somewhat misleading. This variable does nothing specific to enabling use of gluster-block, it merely tells the installer to provision a PVC from the cluster default StorageClass. The docs I'm writing go into more accurate detail on this.

**Review by jarrpa — CHANGES_REQUESTED** [review]
Happy to see the work! :) There's several things I need to get into, however.

**Comment by jarrpa** [comment]
@bmcelvee I understand that the documentation team may still be rather bogged down, so if you like I can take these changes, include them in my PR, and make the changes I'm requesting on top of them myself. I would leave proper accreditation for your work, of course. :)

**Comment by vikram-redhat** [comment]
@jarrpa - were you able to incorporate these changes in your own PR? Or does this still need looking at by @bmcelvee ?

**Comment by bmcelvee** [comment]
@jarrpa Looks like these changes didn't get pulled in. I'm not sure if it's still relevant, but I've updated this PR. PTAL. Thanks!

File: install_config/cluster_metrics.adoc [hunk 8]
```diff
@@ -125,6 +125,10 @@ If you would like to use xref:../install_config/persistent_storage/dynamically_p
 xref:../install_config/cluster_metrics.adoc#metrics-ansible-variables[variable]
 to `dynamic` in the inventory file.
 
+You can use {gluster-native} to enable `gluster-block` Storage
+as a backend for logging with Elasticsearch. For more information, see
```
Comment (jarrpa) [inline install_config/cluster_metrics.adoc]: Looks like this is a copy/paste error from the Logging file. This is likely supposed to talk about Metrics.

Comment (bmcelvee) [reply install_config/cluster_metrics.adoc]: Thanks, @jarrpa. I get that fixed today.

**Pull request merged**